--- origsrc/kdebase-3.5.10/admin/am_edit	2008-08-19 15:27:05.000000000 -0500
+++ src/kdebase-3.5.10/admin/am_edit	2009-07-05 11:49:40.092507800 -0500
@@ -968,7 +968,7 @@ sub tag_KDEINIT()
 
                 $lines .= "${kdeinit}_LDFLAGS = \$(KDE_RPATH) -no-undefined \$(all_libraries)\n";
                 $lines .= "${kdeinit}_LDADD = libkdeinit_${kdeinit}.la\n";
-                $lines .= "${kdeinit}_DEPENDENCIES = libkdeinit_${kdeinit}.la\n";
+                $lines .= "${kdeinit}_DEPENDENCIES = libkdeinit_${kdeinit}.la ${kdeinit}.la\n";
 
                 $target_adds{"${kdeinit}\$(EXEEXT)"} =
                           "\$(${kdeinit}_OBJECTS) \$(${kdeinit}_DEPENDENCIES)\n" .
