--- origsrc/kdebase-3.5.9/kioslave/fish/fish.cpp	2007-05-14 02:55:41.000000000 -0500
+++ src/kdebase-3.5.9/kioslave/fish/fish.cpp	2008-02-21 09:45:55.765625000 -0600
@@ -441,7 +441,7 @@
 #endif
 
         int pgrp = getpid();
-#if defined( _AIX) || defined( __hpux)
+#if defined( _AIX) || defined( __hpux) || defined(__CYGWIN__)
         tcsetpgrp(0, pgrp);
 #else
         ioctl(0, TIOCSPGRP, (char *)&pgrp);
