--- origsrc/kdebase-3.5.9/kate/app/Makefile.am	2005-10-10 10:03:41.000000000 -0500
+++ src/kdebase-3.5.9/kate/app/Makefile.am	2008-02-21 09:45:51.546875000 -0600
@@ -18,11 +18,11 @@
 
 kate_la_SOURCES = katemain.cpp
 kate_la_LIBADD = libkateinterfaces.la
-kate_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+kate_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 kwrite_la_SOURCES = kwritemain.cpp
 kwrite_la_LIBADD   = -lkatepartinterfaces  ../utils/libkateutils.la
-kwrite_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+kwrite_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 METASOURCES = AUTO
 
--- origsrc/kdebase-3.5.9/kcminit/Makefile.am	2006-05-22 13:12:41.000000000 -0500
+++ src/kdebase-3.5.9/kcminit/Makefile.am	2008-02-21 09:45:51.562500000 -0600
@@ -23,11 +23,11 @@
 
 kcminit_la_SOURCES = main.cpp main.skel
 kcminit_la_LIBADD = $(LIB_KDEUI) $(LIB_KUTILS)
-kcminit_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kcminit_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 kcminit_startup_la_SOURCES = main.cpp main.skel
 kcminit_startup_la_LIBADD = $(LIB_KDEUI) $(LIB_KUTILS)
-kcminit_startup_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kcminit_startup_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 AM_CPPFLAGS = $(all_includes)
 METASOURCES = AUTO
--- origsrc/kdebase-3.5.9/kcontrol/access/Makefile.am	2005-09-10 03:25:08.000000000 -0500
+++ src/kdebase-3.5.9/kcontrol/access/Makefile.am	2008-02-21 09:45:51.578125000 -0600
@@ -8,7 +8,7 @@
 kdeinit_LTLIBRARIES = kaccess.la
 
 kaccess_la_SOURCES = kaccess.cpp main.cpp
-kaccess_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kaccess_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kaccess_la_LIBADD = $(LIB_KDEUI)
 
 kcm_access_la_SOURCES = kcmaccess.cpp 
--- origsrc/kdebase-3.5.9/kcontrol/kcontrol/Makefile.am	2005-10-10 10:03:43.000000000 -0500
+++ src/kdebase-3.5.9/kcontrol/kcontrol/Makefile.am	2008-02-21 09:45:52.656250000 -0600
@@ -33,7 +33,7 @@
 	proxywidget.cpp kcrootonly.cpp searchwidget.cpp
 
 kcontrol_la_LIBADD  = $(LIB_KDEUI) $(LIB_KUTILS) $(LIB_KHTML)
-kcontrol_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kcontrol_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 rccontroldir = $(kde_datadir)/kcontrol
 rccontrol_DATA = kcontrolui.rc
--- origsrc/kdebase-3.5.9/kdeprint/kjobviewer/Makefile.am	2005-09-10 03:25:38.000000000 -0500
+++ src/kdebase-3.5.9/kdeprint/kjobviewer/Makefile.am	2008-02-21 09:45:53.921875000 -0600
@@ -9,7 +9,7 @@
 kjobviewer_la_SOURCES = kjobviewer.cpp main.cpp
 kjobviewer_la_METASOURCES = AUTO
 kjobviewer_la_LIBADD = -lkdeprint_management $(LIB_KDEUI)
-kjobviewer_la_LDFLAGS = -module -avoid-version $(all_libraries) $(KDE_RPATH)
+kjobviewer_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 noinst_HEADERS = kjobviewer.h
 
--- origsrc/kdebase-3.5.9/kdeprint/kprinter/Makefile.am	2005-09-10 03:25:39.000000000 -0500
+++ src/kdebase-3.5.9/kdeprint/kprinter/Makefile.am	2008-02-21 09:45:53.937500000 -0600
@@ -8,7 +8,7 @@
 
 kprinter_la_SOURCES = main.cpp printwrapper.cpp
 kprinter_la_LIBADD = -lkdeprint
-kprinter_la_LDFLAGS = -module -avoid-version $(all_libraries) $(KDE_RPATH)
+kprinter_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kprinter_la_METASOURCES = AUTO
 
 noinst_HEADERS = printwrapper.h
--- origsrc/kdebase-3.5.9/khelpcenter/htmlsearch/Makefile.am	2005-09-10 03:25:18.000000000 -0500
+++ src/kdebase-3.5.9/khelpcenter/htmlsearch/Makefile.am	2008-02-21 09:45:54.250000000 -0600
@@ -10,7 +10,7 @@
 lib_LTLIBRARIES = libhtmlsearch.la libkcm_htmlsearch.la
 
 libhtmlsearch_la_SOURCES = htmlsearch.cpp progressdialog.cpp
-libhtmlsearch_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
+libhtmlsearch_la_LDFLAGS = $(all_libraries) -no-undefined
 libhtmlsearch_la_LIBADD = $(LIB_KIO)
 
 libkcm_htmlsearch_la_SOURCES = kcmhtmlsearch.cpp klangcombo.cpp ktagcombobox.cpp
--- origsrc/kdebase-3.5.9/khotkeys/app/Makefile.am	2005-11-19 05:05:32.000000000 -0600
+++ src/kdebase-3.5.9/khotkeys/app/Makefile.am	2008-02-21 09:45:54.250000000 -0600
@@ -9,7 +9,7 @@
 
 kded_khotkeys_la_SOURCES = kded.cpp kded.skel
-kded_khotkeys_la_LIBADD = ../shared/libkhotkeys_shared.la $(LIB_KDECORE)
-kded_khotkeys_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kded_khotkeys_la_LIBADD = ../shared/libkhotkeys_shared.la $(LIB_KDECORE) $(LIB_KDED)
+kded_khotkeys_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 noinst_HEADERS = app.h kded.h
 
--- origsrc/kdebase-3.5.9/kicker/menuext/find/Makefile.am	2005-09-10 03:25:29.000000000 -0500
+++ src/kdebase-3.5.9/kicker/menuext/find/Makefile.am	2008-02-21 09:45:55.109375000 -0600
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_find.la
 
 kickermenu_find_la_SOURCES = findmenu.cpp
-kickermenu_find_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_find_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kickermenu_find_la_LIBADD = $(LIB_KDEUI)
 
 kickermenu_find_la_METASOURCES = AUTO
--- origsrc/kdebase-3.5.9/kicker/menuext/kate/Makefile.am	2007-01-15 05:31:27.000000000 -0600
+++ src/kdebase-3.5.9/kicker/menuext/kate/Makefile.am	2008-02-21 09:45:55.171875000 -0600
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_kate.la
 
 kickermenu_kate_la_SOURCES = katesessionmenu.cpp
-kickermenu_kate_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_kate_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kickermenu_kate_la_LIBADD = $(LIB_KDEUI)
 
 kickermenu_kate_la_METASOURCES = AUTO
--- origsrc/kdebase-3.5.9/kicker/menuext/kdeprint/Makefile.am	2005-09-10 03:25:29.000000000 -0500
+++ src/kdebase-3.5.9/kicker/menuext/kdeprint/Makefile.am	2008-02-21 09:45:55.187500000 -0600
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_kdeprint.la
 
 kickermenu_kdeprint_la_SOURCES = print_mnu.cpp
-kickermenu_kdeprint_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_kdeprint_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kickermenu_kdeprint_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) -lkdeprint
 
 kickermenu_kdeprint_la_METASOURCES = AUTO
--- origsrc/kdebase-3.5.9/kicker/menuext/konq-profiles/Makefile.am	2005-09-10 03:25:29.000000000 -0500
+++ src/kdebase-3.5.9/kicker/menuext/konq-profiles/Makefile.am	2008-02-21 09:45:55.203125000 -0600
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_konqueror.la
 
 kickermenu_konqueror_la_SOURCES = konqy_menu.cpp
-kickermenu_konqueror_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_konqueror_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kickermenu_konqueror_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO)
 
 kickermenu_konqueror_la_METASOURCES = AUTO
--- origsrc/kdebase-3.5.9/kicker/menuext/konsole/Makefile.am	2005-09-10 03:25:29.000000000 -0500
+++ src/kdebase-3.5.9/kicker/menuext/konsole/Makefile.am	2008-02-21 09:45:55.203125000 -0600
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_konsole.la
 
 kickermenu_konsole_la_SOURCES = konsole_mnu.cpp konsolebookmarkhandler.cpp konsolebookmarkmenu.cpp
-kickermenu_konsole_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_konsole_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kickermenu_konsole_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO)
 
 kickermenu_konsole_la_METASOURCES = AUTO
--- origsrc/kdebase-3.5.9/kicker/menuext/prefmenu/Makefile.am	2006-03-17 15:48:32.000000000 -0600
+++ src/kdebase-3.5.9/kicker/menuext/prefmenu/Makefile.am	2008-02-21 09:45:55.250000000 -0600
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_prefmenu.la
 
 kickermenu_prefmenu_la_SOURCES = prefmenu.cpp
-kickermenu_prefmenu_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_prefmenu_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kickermenu_prefmenu_la_LIBADD = $(LIB_KDEUI) $(top_builddir)/kicker/libkicker/libkickermain.la
 
 kickermenu_prefmenu_la_METASOURCES = AUTO
--- origsrc/kdebase-3.5.9/kicker/menuext/remote/Makefile.am	2005-09-10 03:25:29.000000000 -0500
+++ src/kdebase-3.5.9/kicker/menuext/remote/Makefile.am	2008-02-21 09:45:55.296875000 -0600
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_remotemenu.la
 
 kickermenu_remotemenu_la_SOURCES = remotemenu.cpp remotemenu.skel
-kickermenu_remotemenu_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_remotemenu_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kickermenu_remotemenu_la_LIBADD = $(LIB_KDEUI) $(top_builddir)/kicker/libkicker/libkickermain.la
 
 kickermenu_remotemenu_la_METASOURCES = AUTO
--- origsrc/kdebase-3.5.9/kicker/menuext/tom/Makefile.am	2005-09-10 03:25:26.000000000 -0500
+++ src/kdebase-3.5.9/kicker/menuext/tom/Makefile.am	2008-02-21 09:45:55.468750000 -0600
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_tom.la
 
 kickermenu_tom_la_SOURCES = tom.cc
-kickermenu_tom_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_tom_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kickermenu_tom_la_LIBADD = $(LIB_KDEUI)
 #$(top_builddir)/kicker/ui/libkicker_ui.la 
 
--- origsrc/kdebase-3.5.9/kicker/proxy/Makefile.am	2006-10-01 12:31:51.000000000 -0500
+++ src/kdebase-3.5.9/kicker/proxy/Makefile.am	2008-02-21 09:45:55.484375000 -0600
@@ -9,11 +9,11 @@
 
 appletproxy_la_LIBADD = $(LIB_KDEUI) ../libkicker/libkickermain.la
 appletproxy_la_SOURCES = appletproxy.cpp
-appletproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+appletproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module $(KDE_PLUGIN)
 
 extensionproxy_la_LIBADD = $(LIB_KDEUI) ../libkicker/libkickermain.la
 extensionproxy_la_SOURCES = extensionproxy.cpp
-extensionproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+extensionproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module $(KDE_PLUGIN)
 
 extensiondebugger_SOURCES = extensiondebugger.cpp
 extensiondebugger_LDADD = ../libkicker/libkickermain.la
--- origsrc/kdebase-3.5.9/kioslave/home/kdedmodule/Makefile.am	2005-09-10 03:25:37.000000000 -0500
+++ src/kdebase-3.5.9/kioslave/home/kdedmodule/Makefile.am	2008-02-21 09:45:55.906250000 -0600
@@ -4,8 +4,8 @@
 INCLUDES = $(all_includes)
 
 kded_homedirnotify_la_SOURCES = homedirnotify.cpp homedirnotify.skel homedirnotifymodule.cpp homedirnotifymodule.skel
-kded_homedirnotify_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_homedirnotify_la_LIBADD = $(LIB_KSYCOCA)
+kded_homedirnotify_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
+kded_homedirnotify_la_LIBADD = $(LIB_KSYCOCA) $(LIB_KDED)
 
 
 servicesdir = $(kde_servicesdir)/kded
--- origsrc/kdebase-3.5.9/kioslave/media/mediamanager/Makefile.am	2005-09-10 03:25:36.000000000 -0500
+++ src/kdebase-3.5.9/kioslave/media/mediamanager/Makefile.am	2008-02-21 09:45:56.156250000 -0600
@@ -23,8 +23,8 @@
 noinst_LTLIBRARIES = $(LINUXCDPOLLING_LIB) $(HALBACKEND_LIB)
 
 kded_mediamanager_la_SOURCES = mediamanager.cpp mediamanager.skel medialist.cpp backendbase.cpp fstabbackend.cpp removablebackend.cpp mediadirnotify.cpp mediadirnotify.skel
-kded_mediamanager_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_mediamanager_la_LIBADD = $(LIB_KSYCOCA) ../libmediacommon/libmediacommon.la $(HALBACKEND_LIB) $(LINUXCDPOLLING_LIB)
+kded_mediamanager_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
+kded_mediamanager_la_LIBADD = $(LIB_KSYCOCA) $(LIB_KDED) ../libmediacommon/libmediacommon.la $(HALBACKEND_LIB) $(LINUXCDPOLLING_LIB)
 
 
 servicesdir = $(kde_servicesdir)/kded
--- origsrc/kdebase-3.5.9/kioslave/media/medianotifier/Makefile.am	2005-09-10 03:25:36.000000000 -0500
+++ src/kdebase-3.5.9/kioslave/media/medianotifier/Makefile.am	2008-02-21 09:45:56.187500000 -0600
@@ -1,9 +1,9 @@
 INCLUDES = -I$(srcdir)/../libmediacommon -I../libmediacommon $(all_includes)
 
 kde_module_LTLIBRARIES = kded_medianotifier.la
-kded_medianotifier_la_LDFLAGS = -module -avoid-version $(all_libraries)
+kded_medianotifier_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kded_medianotifier_la_LIBADD = ../libmediacommon/libmediacommon.la $(LIB_KDECORE) \
-                               $(LIB_KDEUI) $(LIB_KIO)
+                               $(LIB_KDEUI) $(LIB_KIO) $(LIB_KDED)
 kded_medianotifier_la_SOURCES = medianotifier.cpp medianotifier.skel \
                                 notificationdialog.cpp notificationdialogview.ui
 
--- origsrc/kdebase-3.5.9/kioslave/remote/kdedmodule/Makefile.am	2005-09-10 03:25:37.000000000 -0500
+++ src/kdebase-3.5.9/kioslave/remote/kdedmodule/Makefile.am	2008-02-21 09:45:56.484375000 -0600
@@ -4,8 +4,8 @@
 INCLUDES = $(all_includes)
 
 kded_remotedirnotify_la_SOURCES = remotedirnotify.cpp remotedirnotify.skel remotedirnotifymodule.cpp remotedirnotifymodule.skel
-kded_remotedirnotify_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_remotedirnotify_la_LIBADD = $(LIB_KSYCOCA)
+kded_remotedirnotify_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
+kded_remotedirnotify_la_LIBADD = $(LIB_KSYCOCA) $(LIB_KDED)
 
 
 servicesdir = $(kde_servicesdir)/kded
--- origsrc/kdebase-3.5.9/kioslave/system/kdedmodule/Makefile.am	2005-09-10 03:25:37.000000000 -0500
+++ src/kdebase-3.5.9/kioslave/system/kdedmodule/Makefile.am	2008-02-21 09:45:56.750000000 -0600
@@ -4,8 +4,8 @@
 INCLUDES = $(all_includes)
 
 kded_systemdirnotify_la_SOURCES = systemdirnotify.cpp systemdirnotify.skel systemdirnotifymodule.cpp systemdirnotifymodule.skel
-kded_systemdirnotify_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_systemdirnotify_la_LIBADD = $(LIB_KSYCOCA)
+kded_systemdirnotify_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
+kded_systemdirnotify_la_LIBADD = $(LIB_KSYCOCA) $(LIB_KDED)
 
 
 servicesdir = $(kde_servicesdir)/kded
--- origsrc/kdebase-3.5.9/klipper/Makefile.am	2006-03-17 04:17:30.000000000 -0600
+++ src/kdebase-3.5.9/klipper/Makefile.am	2008-02-21 09:45:56.828125000 -0600
@@ -18,11 +18,11 @@
 
 klipper_la_LIBADD = libklipper_common.la
 klipper_la_SOURCES = dummy.cpp
-klipper_la_LDFLAGS = $(all_libraries) -module -avoid-version
+klipper_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 klipper_panelapplet_la_LIBADD = libklipper_common.la
 klipper_panelapplet_la_SOURCES = applet.cpp applet.skel
-klipper_panelapplet_la_LDFLAGS = $(all_libraries) -module -avoid-version
+klipper_panelapplet_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 xdg_apps_DATA = klipper.desktop
 
--- origsrc/kdebase-3.5.9/kmenuedit/Makefile.am	2006-03-17 04:17:38.000000000 -0600
+++ src/kdebase-3.5.9/kmenuedit/Makefile.am	2008-02-21 09:45:56.843750000 -0600
@@ -16,11 +16,11 @@
 
 kmenuedit_la_SOURCES = main.cpp
 kmenuedit_la_LIBADD = libkmenueditcommon.la
-kmenuedit_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kmenuedit_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 kcontroledit_la_SOURCES = kcontrol_main.cpp
 kcontroledit_la_LIBADD = libkmenueditcommon.la
-kcontroledit_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kcontroledit_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 noinst_HEADERS = kmenuedit.h treeview.h basictab.h khotkeys.h \
 	menufile.h menuinfo.h
--- origsrc/kdebase-3.5.9/konqueror/client/Makefile.am	2005-09-10 03:25:48.000000000 -0500
+++ src/kdebase-3.5.9/konqueror/client/Makefile.am	2008-02-21 09:45:56.921875000 -0600
@@ -5,7 +5,7 @@
 kdeinit_LTLIBRARIES = kfmclient.la
 
 INCLUDES= -I.. $(all_includes)
-kfmclient_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kfmclient_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kfmclient_la_LIBADD =  $(LIB_KIO)
 kfmclient_la_SOURCES = kfmclient.cc KonquerorIface.stub KDesktopIface.stub
 
--- origsrc/kdebase-3.5.9/konqueror/keditbookmarks/Makefile.am	2005-09-10 03:25:49.000000000 -0500
+++ src/kdebase-3.5.9/konqueror/keditbookmarks/Makefile.am	2008-02-21 09:45:56.984375000 -0600
@@ -15,7 +15,7 @@
                             testlink.cpp favicons.cpp updater.cpp exporters.cpp kbookmarknotifier.stub bookmarkinfo.cpp kebsearchline.cpp settings.kcfgc
 kbookmarknotifier_DIR = $(includedir)
 keditbookmarks_la_LIBADD  = $(top_builddir)/libkonq/libkonq.la
-keditbookmarks_la_LDFLAGS = $(all_libraries) -module -avoid-version
+keditbookmarks_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 rcdir = $(kde_datadir)/keditbookmarks
 rc_DATA = keditbookmarksui.rc  keditbookmarks-genui.rc
--- origsrc/kdebase-3.5.9/konqueror/preloader/Makefile.am	2006-03-17 04:17:39.000000000 -0600
+++ src/kdebase-3.5.9/konqueror/preloader/Makefile.am	2008-02-21 09:45:57.046875000 -0600
@@ -4,8 +4,8 @@
 
 INCLUDES= -I.. $(all_includes)
 kded_konqy_preloader_la_SOURCES = preloader.cc preloader.skel
-kded_konqy_preloader_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_konqy_preloader_la_LIBADD = ../libkonqueror_intern.la $(LIB_KSYCOCA) $(LIB_KDECORE)
+kded_konqy_preloader_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
+kded_konqy_preloader_la_LIBADD = ../libkonqueror_intern.la $(LIB_KDED) $(LIB_KSYCOCA) $(LIB_KDECORE)
 
 METASOURCES = AUTO
 
--- origsrc/kdebase-3.5.9/konsole/konsole/Makefile.am	2006-03-17 04:17:43.000000000 -0600
+++ src/kdebase-3.5.9/konsole/konsole/Makefile.am	2008-02-21 09:45:57.546875000 -0600
@@ -33,15 +33,15 @@
 
 # kwrited kded module
 kded_kwrited_la_SOURCES = kwrited.cpp TEPty.cpp kwrited.skel
-kded_kwrited_la_LIBADD  = $(LIB_KDECORE) 
-kded_kwrited_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kded_kwrited_la_LIBADD  = $(LIB_KDED) $(LIB_KDECORE)
+kded_kwrited_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 # konsole kdeinit module
 konsole_la_SOURCES = TEPty.cpp BlockArray.cpp main.cpp konsole.cpp schema.cpp session.cpp TEWidget.cpp TEmuVt102.cpp \
      TEScreen.cpp TEmulation.cpp TEHistory.cpp keytrans.cpp konsoleiface.skel sessioniface.skel \
      konsole_wcwidth.cpp konsolebookmarkhandler.cpp konsolebookmarkmenu.cpp \
      zmodem_dialog.cpp printsettings.cpp
-konsole_la_LDFLAGS = $(all_libraries) -module -avoid-version
+konsole_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 konsole_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) $(LIB_KDEPRINT) $(LIBUTIL) $(XTESTLIB) $(LIB_XRENDER)
 
 # kcmkonsole_SOURCES = kcmkonsole.cpp schema.cpp
--- origsrc/kdebase-3.5.9/ksmserver/Makefile.am	2006-01-19 11:01:08.000000000 -0600
+++ src/kdebase-3.5.9/ksmserver/Makefile.am	2008-02-21 09:45:57.609375000 -0600
@@ -30,7 +30,7 @@
 	legacy.cpp startup.cpp shutdown.cpp client.cpp \
 	KSMServerInterface.skel server.skel
 
-ksmserver_la_LDFLAGS = $(all_libraries) -avoid-version -module
+ksmserver_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 ksmserver_la_LIBADD = ../kdmlib/libdmctl.la $(LIB_KDEUI)
 
 picsdir = $(kde_datadir)/ksmserver/pics
--- origsrc/kdebase-3.5.9/kwin/Makefile.am	2005-09-10 03:26:03.000000000 -0500
+++ src/kdebase-3.5.9/kwin/Makefile.am	2008-02-21 09:45:58.171875000 -0600
@@ -18,7 +18,7 @@
 	geometry.cpp rules.cpp
 
 kwin_la_LIBADD = $(LIB_KDEUI) lib/libkdecorations.la
-kwin_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kwin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 include_HEADERS = KWinInterface.h
 
--- origsrc/kdebase-3.5.9/kxkb/Makefile.am	2006-03-17 04:17:41.000000000 -0600
+++ src/kdebase-3.5.9/kxkb/Makefile.am	2008-02-21 09:45:58.828125000 -0600
@@ -13,13 +13,13 @@
 kcm_keyboard_la_SOURCES = extension.cpp x11helper.cpp rules.cpp kxkbconfig.cpp pixmap.cpp \
 	kcmlayout.cpp kcmlayoutwidget.ui kcmmisc.cpp kcmmiscwidget.ui
-kcm_keyboard_la_LDFLAGS  = $(all_libraries) -module -avoid-version -no-undefined
+kcm_keyboard_la_LDFLAGS  = $(all_libraries) -module $(KDE_PLUGIN)
 kcm_keyboard_la_LIBADD = -lxkbfile $(XTESTLIB) $(LIB_KIO)
 
 #kxkb_la_includedir = .
 #kxkb_la_include_HEADERS = kxkb.h
 kxkb_la_SOURCES = extension.cpp x11helper.cpp rules.cpp kxkbconfig.cpp pixmap.cpp \
 	layoutmap.cpp kxkb.cpp kxkbtraywindow.cpp kxkb.skel
-kxkb_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kxkb_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 kxkb_la_LIBADD = -lxkbfile -lX11 -lXext $(LIB_KDEUI)
 
 xdg_apps_DATA = keyboard.desktop keyboard_layout.desktop
--- origsrc/kdebase-3.5.9/libkonq/favicons/Makefile.am	2005-09-10 03:25:53.000000000 -0500
+++ src/kdebase-3.5.9/libkonq/favicons/Makefile.am	2008-02-21 09:45:58.875000000 -0600
@@ -2,8 +2,8 @@
 
 INCLUDES = $(all_includes)
 kded_favicons_la_SOURCES = favicons.cpp favicons.skel
-kded_favicons_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_favicons_la_LIBADD = $(LIB_KSYCOCA)
+kded_favicons_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
+kded_favicons_la_LIBADD = $(LIB_KDED) $(LIB_KSYCOCA)
 
 METASOURCES = AUTO
 
--- origsrc/kdebase-3.5.9/nsplugins/Makefile.am	2007-10-08 04:51:24.000000000 -0500
+++ src/kdebase-3.5.9/nsplugins/Makefile.am	2008-02-21 09:45:58.953125000 -0600
@@ -24,6 +24,7 @@ kcm_nsplugins_la_LIBADD = $(LIB_KDECORE)
 
 noinst_LTLIBRARIES = libpluginpaths.la
 libpluginpaths_la_SOURCES = plugin_paths.cpp
+libpluginpaths_la_LIBADD = $(LIB_KDECORE)
 
 xdg_apps_DATA = khtml_plugins.desktop
 
--- origsrc/kdebase-3.5.9/nsplugins/wrapper/Makefile.am	2005-09-10 03:25:39.000000000 -0500
+++ src/kdebase-3.5.9/nsplugins/wrapper/Makefile.am	2008-02-21 09:45:58.953125000 -0600
@@ -1,4 +1,5 @@
 lib_LTLIBRARIES = libnswrapper.la
 libnswrapper_la_SOURCES = wrapper.c
+libnswrapper_la_LDFLAGS = -no-undefined
 INCLUDES = -I$(top_srcdir)/nsplugins
 
