--- origsrc/kdebase-3.5.9/kicker/extensions/kasbar/kasloaditem.cpp	2005-09-10 03:25:29.000000000 -0500
+++ src/kdebase-3.5.9/kicker/extensions/kasbar/kasloaditem.cpp	2008-02-21 09:45:55.000000000 -0600
@@ -30,6 +30,45 @@
 #include "kasloaditem.h"
 #include "kasloaditem.moc"
 
+/*
+ * getloadavg() for systems that don't have that library call.
+ * http://gehenna.pell.portland.or.us/~orc/Code/postoffice/postoffice-1.3.2/getloadavg.c
+ */
+#if defined(__CYGWIN__)
+
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+extern "C" {
+int
+getloadavg(double *la, int size)
+{
+    char bfr[256];
+    int fd, bufsiz, ret=-1, i;
+    float vla[3];
+
+    if ( (fd = open("/proc/loadavg", O_RDONLY)) != -1) {
+
+	if ( (bufsiz = read(fd, bfr, sizeof bfr)) > 0) {
+	    bfr[bufsiz] = 0;
+	    ret = sscanf(bfr, "%f %f %f ", &vla[0], &vla[1], &vla[2]);
+
+	    if (size < ret) ret = size;
+
+	    for (i=0; i < ret; i++)
+		la[i] = vla[i];
+	}
+	close(fd);
+    }
+    return ret;
+}
+}
+#endif
+
+
 KasLoadItem::KasLoadItem( KasBar *parent )
     : KasItem( parent )
 {
