--- origsrc/kdebase-3.5.10/kdesu/kdesud/kdesud.cpp	2007-05-14 02:55:53.000000000 -0500
+++ src/kdebase-3.5.10/kdesu/kdesud/kdesud.cpp	2009-05-04 20:03:18.296750000 -0500
@@ -388,7 +388,7 @@ int main(int argc, char *argv[])
                 // Accept new connection
                 int fd;
                 addrlen = 64;
-                fd = accept(sockfd, (struct sockaddr *) &clientname, &addrlen);
+                fd = accept(sockfd, (struct sockaddr *) &clientname, (socklen_t *) &addrlen);
                 if (fd < 0) 
                 {
                     kdError(1205) << "accept():" << ERR << "\n";
--- origsrc/kdebase-3.5.10/kdesu/kdesud/secure.cpp	2005-09-10 03:26:13.000000000 -0500
+++ src/kdebase-3.5.10/kdesu/kdesud/secure.cpp	2009-05-04 20:05:14.031125000 -0500
@@ -32,7 +32,7 @@
 SocketSecurity::SocketSecurity(int sockfd)
 {
     ksocklen_t len = sizeof(struct ucred);
-    if (getsockopt(sockfd, SOL_SOCKET, SO_PEERCRED, &cred, &len) < 0) {
+    if (getsockopt(sockfd, SOL_SOCKET, SO_PEERCRED, &cred, (socklen_t *) &len) < 0) {
 	kdError() << "getsockopt(SO_PEERCRED) " << perror << endl;
 	return;
     }
