--- origsrc/kdebase-3.5.9/kdesu/kdesud/kdesud.cpp	2007-05-14 02:55:53.000000000 -0500
+++ src/kdebase-3.5.9/kdesu/kdesud/kdesud.cpp	2008-02-21 09:45:54.046875000 -0600
@@ -81,6 +81,10 @@
                      + strlen ((ptr)->sun_path))   
 #endif
 
+#ifdef __CYGWIN__
+typedef unsigned ksocklen_t;
+#endif
+
 #define ERR strerror(errno)
 
 // Globals
@@ -318,7 +322,11 @@
 
     // Main execution loop 
 
+#ifdef __CYGWIN__
+    int addrlen;
+#else
     ksocklen_t addrlen;
+#endif
     struct sockaddr_un clientname;
 
     fd_set tmp_fds, active_fds;
--- origsrc/kdebase-3.5.9/kdesu/kdesud/secure.cpp	2005-09-10 03:26:13.000000000 -0500
+++ src/kdebase-3.5.9/kdesu/kdesud/secure.cpp	2008-02-21 09:45:54.062500000 -0600
@@ -31,7 +31,11 @@
 
 SocketSecurity::SocketSecurity(int sockfd)
 {
+#ifdef __CYGWIN__
+    int len = sizeof(struct ucred);
+#else
     ksocklen_t len = sizeof(struct ucred);
+#endif
     if (getsockopt(sockfd, SOL_SOCKET, SO_PEERCRED, &cred, &len) < 0) {
 	kdError() << "getsockopt(SO_PEERCRED) " << perror << endl;
 	return;
