--- origsrc/kdebase-3.5.9/kcontrol/nics/nic.cpp	2007-01-15 05:31:40.000000000 -0600
+++ src/kdebase-3.5.9/kcontrol/nics/nic.cpp	2008-02-21 09:45:53.421875000 -0600
@@ -193,8 +193,10 @@
 
          if ((flags & IFF_BROADCAST) == IFF_BROADCAST)
             tmp->type=i18n("Broadcast");
+#ifdef IFF_POINTOPOINT
          else if ((flags & IFF_POINTOPOINT) == IFF_POINTOPOINT)
             tmp->type=i18n("Point to Point");
+#endif
 #ifndef _AIX
          else if ((flags & IFF_MULTICAST) == IFF_MULTICAST)
             tmp->type=i18n("Multicast");
