--- origsrc/kdebase-3.5.9/kdm/backend/dm.h	2005-09-10 03:26:12.000000000 -0500
+++ src/kdebase-3.5.9/kdm/backend/dm.h	2008-02-21 09:45:54.093750000 -0600
@@ -105,7 +105,7 @@
 #endif
 
 #include <utmp.h>
-#ifdef HAVE_UTMPX
+#if defined(HAVE_UTMPX) && !defined(__CYGWIN__)
 # include <utmpx.h>
 # define STRUCTUTMP struct utmpx
 # define UTMPNAME utmpxname
--- origsrc/kdebase-3.5.9/kdm/backend/sessreg.c	2006-03-17 04:17:47.000000000 -0600
+++ src/kdebase-3.5.9/kdm/backend/sessreg.c	2008-02-21 09:45:54.109375000 -0600
@@ -275,7 +275,7 @@
 		ll.ll_time = ut_ent.ut_time;
 		memcpy( ll.ll_line, ut_ent.ut_line, sizeof(ll.ll_line) );
 		memcpy( ll.ll_host, ut_ent.ut_host, sizeof(ll.ll_host) );
-# ifdef HAVE_UTMPX
+# if defined(HAVE_UTMPX) && !defined(__CYGWIN__)
 		updlastlogx( LLOG_FILE, uid, &ll );
 # else
 		if ((llog = open( LLOG_FILE, O_RDWR )) < 0)
